roda-core/roda-core/src/main/java/org/roda/core/plugins/plugins/internal/synchronization/SyncBundleHelper.java

```diff
@@ -145,4 +161,98 @@ public static String calculateEntityTopHash(String entity) throws GenericExcepti
+   * 
+   * @param bundleState
+   *          {@link BundleState}.
+   * @return {@link Map}.
+   */
+  private static Map<Path, Class<? extends IsIndexed>> createLocalInstanceListPaths(final BundleState bundleState) {
+    final HashMap<Path, Class<? extends IsIndexed>> localInstanceListsMap = new HashMap<>();
+    final Path aipListPath = Paths.get(bundleState.getDestinationPath())
+      .resolve(bundleState.getEntitiesBundle().getAipFileName() + ".json");
+    localInstanceListsMap.put(aipListPath, IndexedAIP.class);
+
+    final Path dipListPath = Paths.get(bundleState.getDestinationPath())
+      .resolve(bundleState.getEntitiesBundle().getDipFileName() + ".json");
+    localInstanceListsMap.put(dipListPath, IndexedDIP.class);
+
+    // Not necessary
+    // final Path jobListPath = Paths.get(bundleState.getDestinationPath())
+    // .resolve(bundleState.getEntitiesBundle().getJobFileName() + ".json");
+    // localInstanceListsMap.put(jobListPath, Job.class);
+
+    // final Path preservationAgentListPath =
+    // Paths.get(bundleState.getDestinationPath())
+    // .resolve(bundleState.getEntitiesBundle().getPreservationAgentFileName() +
+    // ".json");
+    // localInstanceListsMap.put(preservationAgentListPath,
+    // IndexedPreservationAgent.class);
+    //
+    // final Path repositoryEventListPath =
+    // Paths.get(bundleState.getDestinationPath())
+    // .resolve(bundleState.getEntitiesBundle().getRepositoryEventFileName() +
+    // ".json");
+    // localInstanceListsMap.put(repositoryEventListPath,
+    // IndexedPreservationEvent.class);
+
+    final Path riskListPath = Paths.get(bundleState.getDestinationPath())
+      .resolve(bundleState.getEntitiesBundle().getRiskFileName() + ".json");
+    localInstanceListsMap.put(riskListPath, RiskIncidence.class);
+
+    return localInstanceListsMap;
+  }
+
+  /**
+   * Write the List in the file to bundle.
+   * 
+   * @param destinationPath
+   *          {@link Path}.
+   * @param indexedClass
+   *          {@link Class<? extends IsIndexed>}.
+   * @throws GenericException
+   *           if some error occurs.
+   * @throws IOException
+   *           if some i/o error occurs.
+   */
+  private static void createLocalInstanceList(final Path destinationPath, final Class<? extends IsIndexed> indexedClass)
+    throws GenericException, IOException {
+    final List<String> list = new ArrayList<>();
+    final IndexService index = RodaCoreFactory.getIndexService();
+    final Filter filter = new Filter();
+    if (indexedClass == IndexedPreservationEvent.class) {
+      filter.add(new SimpleFilterParameter(RodaConstants.PRESERVATION_EVENT_OBJECT_CLASS,
+        IndexedPreservationEvent.PreservationMetadataEventClass.REPOSITORY.toString()));
+    }
+
+    try (IterableIndexResult<? extends IsIndexed> result = index.findAll(indexedClass, filter, true,
+      Collections.singletonList(RodaConstants.INDEX_UUID))) {
+      result.forEach(indexed -> list.add(indexed.getId()));
```

Comment:
There is a potential memory problem here. Please find another solution. 
See suggestion in CentralEntitiesJsonUtils class review